lens/utils/_netflow/nf_cplex.py

```
@@ -13,212 +13,216 @@
+            return sorted(self._materialIdxLookup, key=self._materialIdxLookup.__getitem__)
+
+        def getUpperBounds(self):
+            return self._ub.copy()
+
+        def getLowerBounds(self):
+            return self._lb.copy()
+
+        def getObjective(self):
+            return self._objective.copy()
+
+        def buildEqConst(self):
+            if self._eqConstBuilt:
+                raise Exception("Equality constraints already built.")
+
+            nMaterials = len(self._materialCoeffs)
+            nFlows = len(self._flows)
+            A = np.zeros((nMaterials, nFlows))
+
+            # avoid creating duplicate constraints
+            self._materialIdxLookup = {}
+            for materialIdx, (material, pairs) in enumerate(sorted(self._materialCoeffs.viewitems())):
+                self._materialIdxLookup[material] = materialIdx
+                for pair in pairs:
+                    A[materialIdx, pair[1]] = pair[0]
+            A_coo = coo_matrix(A)
+            row = A_coo.row.astype(np.int32).astype(np.int32)
+            col = A_coo.col.astype(np.int32).astype(np.int32)
+            data = A_coo.data.astype(np.float64).astype(np.float64)
+
+            # set solver constraints
+            self._model.linear_constraints.add(rhs=np.zeros(nMaterials), senses='E'*nMaterials)
+            self._model.linear_constraints.set_coefficients(zip(row.tolist(), col.tolist(), data.tolist()))
+
+            self._eqConstBuilt = True
+
+        def _solve(self):
+            if self._solved:
+                return
 
-        self._model.solve()
+            if self._maximize:
+                self._model.objective.set_sense(self._model.objective.sense.maximize)
+            else:
+                self._model.objective.set_sense(self._model.objective.sense.minimize)
 
-        self._solved = True
+            self._model.solve()
+
+            self._solved = True
+except:
```

BTW, one should never catch all exceptions unless the code just logs and rethrows the exception. This blocks things like ^C interrupts.

---

Ah yes, updated to only catch `ImportError`.